src/praisonai-agents/tests/integration/test_hermes_filter_integration.py

```
@@ -0,0 +1,262 @@
+        tool = registry.get("search")
+        assert callable(tool)  # Should be the function, not the MockTool
+    
+    @patch.dict(os.environ, {"CI": "true"})
+    def test_strict_mode_in_ci(self):
+        """Test strict failure mode in CI environment."""
+        registry = ToolRegistry()
+        registry.register(MockTool("available_tool"))
+        
+        # Should fail strictly in CI when unknown tools are specified
+        with patch.dict(os.environ, {"HERMES_ONLY_TOOLS": "available_tool,unknown_tool"}):
+            with pytest.raises(ValueError, match="Unknown tools.*unknown_tool"):
```

Ruff (0.15.13)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)